docs/tcpAgent.md

```diff
@@ -0,0 +1,140 @@
+Internally, this is a two-step process: 1) connect to an HTTP(S) port to retrieve the connection information, and 2) extract that information and connect to the TCP port.
+
+Before invoking this command you must download the "agent.jar" file.
+The correct version can be obtained from your Jenkins server at something like "https://myjenkins.example.com/jnlpJars/agent.jar".
+These operations can be combined in a script, cron job, or service.
+
+A typical example of this command, as shown in the above image, is
+``` 
+java -jar agent.jar \
+  -jnlpUrl <jnlp url> \
+  -secret <secret key> \
+  -workDir <work directory> 
+```
+
+### Connect directly to TCP port
+With the latest versions of the Remoting library (since TODO), it is possible to connect directly to the TCP port. 
+This skips the download of the connection information file, described in step 1) above. 
+To use this you must provide some of the information that the downloaded file would contain.
+
+You must provide the agent name. 
+This is easily obtained from the agent status page or other places.
+
+You need to specify the host and port, in a typical `HOST:PORT` fashion. 
+The HOST can be whatever name or address resolves to the server. 
+The PORT value can be determined by examining the JNLP file or looking at the Agents section of the server Configure Global Security page.
+
+Each Jenkins instance has its own [Instance Identity](https://wiki.jenkins.io/display/JENKINS/Instance+Identity). 
+The agent needs this key to complete the connection. 
+You can obtain this value via the script console or the Instance Identity page.
+
+Using the script console, execute the command
+```
+hudson.remoting.Base64.encode(org.jenkinsci.main.modules.instance_identity.InstanceIdentity.get().getPublic().getEncoded())
+```
+Capture the result to pass into the agent invocation.
+
+You can also capture the value by visiting the Instance Identity page, at something like "https://myjenkins.example.com/instance-identity". 
+If you use this mechanism you must stitch it back together into a single, complete value from its multi-line display representation.
+
+This mechanism requires a download of the `agent.jar`, as described for "Download JNLP file".
+
+Once all the prerequisite files and data have been obtained, the agent can be launched with a command like this
+```
+java -cp agent.jar hudson.remoting.jnlp.Main \
+  -headless \
+  -workDir <work directory> \
+  -direct <HOST:PORT> \
+  -protocols JNLP4-connect \
+  -instanceIdentity <instance identity> \
+  <secretString> <agentName>
+```
+The "-protocols" parameter is optional, but is useful to limit the agent to protocols the server supports. 
+The only currently supported and recommended protocol is "JNLP4-connect".
+
+Note that this mechanism uses a different JAR entry point than used for "Download JNLP file". 
+The parameters available and the default behavior may be different between the entry points.
```

Comment:
Do you think it would be worthwhile to note here to see the "@" parameter section later for a more secure way of launching the agent without having the secrets in the process list? Though it's not exactly far away from here, it might be useful?

Reply:
I don't think so. The documentation for the "@" parameters is close enough that referencing it here and in the prior subsection would just add clutter. And it's quite debatable that format is more secure.